### PR TITLE
containers/ideas: delete idea

### PR DESCRIPTION
**Pull request by fuzzylogic2000**
depending on: https://github.com/liqd/adhocracy-plus/pull/1395

No error handling so far, but should be ok as button only shown when deletion allowed.

### CODE REVIEW

File: containers/Ideas/Idea.js [hunk 1]
```diff
@@ -34,13 +34,13 @@ export const Idea = (props) => {
...
       action: () => toggleDeleteModal(),
-      isAllowed: params.has_changing_permission
+      isAllowed: params.has_deleting_permission
     },
     {
       title: 'Report',
       icon: 'flag',
       action: () => console.log('Report'),
-      isFirst: !params.has_changing_permission,
+      isFirst: !params.has_changing_permission && !params.has_deleting_permission,
```
Comment (khamui) [inline containers/Ideas/Idea.js:43]: nice!!

**Review by khamui — COMMENTED** [review]
pretty cool! Deleting works!

Somehow the recent deleted idea is still listed on the project page. Do we want to tackle that now? Or shall I create an issue?

**Comment by fuzzylogic2000** [comment]
> Somehow the recent deleted idea is still listed on the project page. Do we want to tackle that now? Or shall I create an issue?

I needed quite a while to get that working and it was fine locally for me. And it should work, right? I don't use the ideaProject page from the navigator stack, but go to it and by that reload the page?!

**Comment by khamui** [comment]
> > Somehow the recent deleted idea is still listed on the project page. Do we want to tackle that now? Or shall I create an issue?
> 
> I needed quite a while to get that working and it was fine locally for me. And it should work, right? I don't use the ideaProject page from the navigator stack, but go to it and by that reload the page?!

yes it should work by using `goBack()`, but it might be related with that listener `focus` in the project page, which somehow doesn't update the list. But if it worked for you we can merge and test on dev?

Not sure if that helps:
https://github.com/react-navigation/react-navigation.github.io/issues/191#issuecomment-641018588

File: containers/Ideas/Idea.js [hunk 2]
```diff
@@ -114,6 +114,16 @@ export const Idea = (props) => {
       setVoteDown(0);
     }
   };
+
+  const deleteIdea = () => {
+    AsyncStorage.getItem('authToken')
+      .then((token) => API.deleteIdea(moduleId, params.pk, token))
+      .then(() => {
+        Alert.alert('Your idea was deleted.', 'Thank you for participating!',  [{ text: 'Ok' }]);
+        props.navigation.navigate('IdeaProject');
```
Comment (fuzzylogic2000) [inline containers/Ideas/Idea.js:123]: @khamui But I don't use goBack(), but this. goBack didn't reload the list, but for me this did. Maybe @Rineee also needs to try and decide if it works or not. :p

Comment (khamui) [reply containers/Ideas/Idea.js:123]: @fuzzylogic2000 oh I tried an old version locally. With the new version all is good and working fine!

**Pull request merged**